toolchain/check/context.cpp

```diff
@@ -112,6 +114,41 @@ auto Context::NoteIncompleteClass(SemIR::ClassId class_id,
   }
 }
 
+auto Context::AddPackageImports(Parse::Node import_node,
+                                IdentifierId package_id,
+                                llvm::ArrayRef<const SemIR::File*> sem_irs,
+                                bool has_failure) -> void {
+  CARBON_CHECK(has_failure || !sem_irs.empty())
+      << "There should be either a failed import or at least one IR.";
+
+  auto name_id = SemIR::NameId::ForIdentifier(package_id);
+
+  SemIR::CrossReferenceIRId first_id(cross_reference_irs().size());
+  for (const auto* sem_ir : sem_irs) {
+    cross_reference_irs().Add(sem_ir);
+  }
+  if (has_failure) {
+    cross_reference_irs().Add(nullptr);
+  }
+  SemIR::CrossReferenceIRId last_id(cross_reference_irs().size() - 1);
+
+  auto inst_id = AddInst(SemIR::Import{.parse_node = import_node,
+                                       .first_cross_reference_ir_id = first_id,
+                                       .last_cross_reference_ir_id = last_id});
```

Comment:
We switched to giving things that can be found by name lookup (namespaces, functions) a type in order to reduce the need to special-case them elsewhere and avoid crashes when downstream checking code tries to look at the type of a thing it found by name lookup. I think we should do the same here. Maybe we could use `BuiltinNamespaceType` for package names too, given that the design says that a package name is a kind of namespace. This'd also fix the SemIR printing, where you have references to a `%import` that is never declared (we don't print a name for untyped nodes because we don't expect them to be referenced).